packages/hardhat/contracts/strategy/CrabOTC.sol

```diff
@@ -0,0 +1,208 @@
+//SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import {ICrabStrategyV2} from "../interfaces/ICrabStrategyV2.sol";
+import {IWETH9} from "../interfaces/IWETH9.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {EIP712} from "@openzeppelin/contracts/drafts/EIP712.sol";
+import {ECDSA} from "@openzeppelin/contracts/cryptography/ECDSA.sol";
+import {StrategyMath} from "./base/StrategyMath.sol";
+
+contract CrabOTC is EIP712 {
+    using StrategyMath for uint256;
+    using Address for address payable;
+
+    address public immutable crab;
+    address public immutable weth;
+    address public immutable controller;
+    address public immutable wPowerPerp;
+    mapping(address => mapping(uint256 => bool)) public nonces;
+
+    struct Order {
+        address initiator;
+        address trader;
+        uint256 quantity;
+        uint256 price;
+        bool isBuying;
+        uint256 expiry;
+        uint256 nonce;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    event DepositOTC(
+        address indexed depositor,
+        uint256 crabAmount,
+        uint256 wSqueethAmount,
+        uint256 depositedAmount,
+        uint256 executedPrice,
+        address trader
+    );
+
+    event WithdrawOTC(
+        address indexed depositor,
+        uint256 crabAmount,
+        uint256 ethSent,
+        uint256 wSqueethAmount,
+        uint256 executedPrice,
+        address trader
+    );
+
+    bytes32 private constant _CRAB_BALANCE_TYPEHASH =
+        keccak256(
+            "Order(address initiator,address trader,uint256 quantity,uint256 price,bool isBuying,uint256 expiry,uint256 nonce)"
+        );
+
+    constructor(address _crab) EIP712("CrabOTC", "2") {
+        require(_crab != address(0), "Invalid crab address");
+
+        crab = _crab;
+        weth = ICrabStrategyV2(_crab).weth();
+        controller = ICrabStrategyV2(_crab).powerTokenController();
+        wPowerPerp = ICrabStrategyV2(_crab).wPowerPerp();
+    }
+
+    /**
+     * @notice set nonce to true
+     * @param _nonce the number to be set true
+     */
+    function setNonceTrue(uint256 _nonce) external {
+        nonces[msg.sender][_nonce] = true;
+    }
+
+    function deposit(
+        uint256 _totalEth,
+        uint256 _minPrice,
+        Order memory _order
+    ) external payable {
+        _verifyOrder(_order, true, _minPrice);
+        uint256 depositedEth = msg.value;
+
+        uint256 wSqueethQuantity = _getWSqueethToMint(_totalEth);
+        require(_order.quantity >= wSqueethQuantity, "Order quantity is less than needed");
```

Comment:
User cant change the quantity after the MM has signed the tx. That would invalidate the signature. We needed >= because frontend does not calculate squeeth to mint with precision